File: 1954-replace-all-digits-with-characters/1954-replace-all-digits-with-characters.java
class Solution {
    public String replaceDigits(String s) {
        char[] words = s.toCharArray();

        for (int i = 1; i < s.length(); i += 2) {
            words[i] = (char) ((int) words[i - 1] + words[i] - '0');
        }

        return new String(words);
    }
}

File: 1954-replace-all-digits-with-characters/1954-replace-all-digits-with-characters.rs
impl Solution {
    pub fn replace_digits(s: String) -> String {
        let mut res = vec![];

        for chunk in s.as_bytes().chunks(2) {
            res.push(chunk[0]);
            if chunk.len() == 2 {
                res.push(chunk[0] + (chunk[1] - b'0'));
            }
        }

        String::from_utf8(res).unwrap_or_default()
    }
}